### ISSUE TITLE
Condition connections are broken

### ISSUE THREAD

**Issue by maxfischer2781** [MaineKuehn/usim#5]
The ``Connective`` used in connecting ``Condition``s (``a & b & c | d``) does not properly unpack on chaining. The issue is the second set in ``Connective.__init__``, which does not unpack:

```python3
        self._children = tuple(
            set(condition for condition in conditions if not isinstance(condition, self.__class__)) |
            set(condition._children for condition in conditions if isinstance(condition, self.__class__))
        )
```

The ``_children`` are "unpacked" to ``((a, b), c)`` instead of ``(a, b, c)``.

**Comment by maxfischer2781** [comment]
Suggested change:

* Add unittest to detect the error.
* Move the unpacking code to the respective subclasses. For example, ``All`` can overload the ``&`` operator to do the unpacking:

```python3
    def __and__(self: All, other) -> 'Condition':
        if isinstance(other, All):
            return All(*self._children, *other._children)
        return All(*self._children, other)
```

**Comment by maxfischer2781** [comment]
Fixed with #7.
